src/components/app.cjsx

```diff
@@ -97,18 +116,22 @@ module.exports = React.createClass
         <div className="container-fluid">
           <div className="navbar-header">
             <BS.ButtonToolbar className="navbar-btn">
-              <a href="/exercises/new" className="btn btn-success">New Blank Exercise</a>
-              <AsyncButton
-                bsStyle='info'
-                onClick={@saveExercise}
-                disabled={isWorking}
-                isWaiting={ExerciseStore.isSaving(id)}
-                waitingText='Saving...'
-                isFailed={ExerciseStore.isFailed(id)}
-                >
-                Save Draft
-              </AsyncButton>
-              { if not ExerciseStore.isNew(id)
+              <a href="/exercises" onClick={@onReset} className="btn btn-danger">Reset</a>
```

Comment:
It seemed the intent was to prompt before losing unsaved changes and that sounded to me like a `onpageunload` event (or a hook in react-router if you're not actually leaving the JS). Does the prompt still happen when you tab to the link and press space (instead of actually clicking)?